Fix failing TestAwsS3SSECustomKey tests

## Describe the bug

Noiced that on CICD fails TestAwsS3SSECustomKey test

## Reproducing bugs

On running CICD jobs, fails TestAwsS3SSECustomKey

```
2026-01-29T23:58:58.3767305Z time=2026-01-29UTC23:49:06Z level=stderr prefix=../../../../002 tf-path=terraform msg=╵
2026-01-29T23:58:58.3770108Z     integration_s3_encryption_test.go:103: Failed to run *** command 'terragrunt apply -auto-approve --backend-bootstrap --non-interactive --config /tmp/TestAwsS3SSECustomKey4004142932/002/terragrunt.hcl --working-dir /tmp/TestAwsS3SSECustomKey4004142932/001/fixtures/s3-encryption/custom-key' due to error: util.ProcessExecutionError Failed to execute "terraform apply -auto-approve" in /tmp/TestAwsS3SSECustomKey4004142932/001/fixtures/s3-encryption/custom-key/.terragrunt-cache/XSkqWcbClMoCc3TD15l6re4Dt4Q/FfOEuVHs_4duVz4jR2poCwSCDEc
2026-01-29T23:58:58.3772435Z         [31m╷[0m[0m
2026-01-29T23:58:58.3773029Z         [31m│[0m [0m[1m[31mError: [0m[0m[1mRequired plugins are not installed[0m
2026-01-29T23:58:58.3773575Z         [31m│[0m [0m
2026-01-29T23:58:58.3774194Z         [31m│[0m [0m[0mThe installed provider plugins are not consistent with the packages
2026-01-29T23:58:58.3774895Z         [31m│[0m [0mselected in the dependency lock file:
2026-01-29T23:58:58.3775890Z         [31m│[0m [0m  - registry.opentofu.org/hashicorp/null: there is no package for registry.opentofu.org/hashicorp/null 3.2.4 cached in .terraform/providers
2026-01-29T23:58:58.3776708Z         [31m│[0m [0m
2026-01-29T23:58:58.3777283Z         [31m│[0m [0mTerraform uses external plugins to integrate with a variety of different
2026-01-29T23:58:58.3778171Z         [31m│[0m [0minfrastructure services. To download the plugins required for this
2026-01-29T23:58:58.3778766Z         [31m│[0m [0mconfiguration, run:
2026-01-29T23:58:58.3779225Z         [31m│[0m [0m  terraform init
2026-01-29T23:58:58.3779639Z         [31m╵[0m[0m
2026-01-29T23:58:58.3779957Z         
2026-01-29T23:58:58.3780249Z         exit status 1
2026-01-29T23:58:58.3780751Z         /home/runner/work/terragrunt/terragrunt/internal/shell/run_cmd.go:152 (0x1a7bf18)
2026-01-29T23:58:58.3781377Z         	RunCommandWithOutput.func1: return errors.New(err)
2026-01-29T23:58:58.3781989Z         /home/runner/work/terragrunt/terragrunt/internal/telemetry/meter.go:76 (0x1291e1c)
2026-01-29T23:58:58.3782562Z         	(*Meter).Time: return fn(ctx)
2026-01-29T23:58:58.3783277Z         /home/runner/work/terragrunt/terragrunt/internal/telemetry/telemeter.go:59 (0x1a7ae7e)
2026-01-29T23:58:58.3784070Z         	RunCommandWithOutput.(*Telemeter).Collect.func2: return tlm.Time(ctx, name, attrs, fn)
2026-01-29T23:58:58.3784838Z         /home/runner/work/terragrunt/terragrunt/internal/telemetry/tracer.go:191 (0x1294995)
2026-01-29T23:58:58.3785430Z         	(*Tracer).Trace: return fn(ctx)
2026-01-29T23:58:58.3786004Z         /home/runner/work/terragrunt/terragrunt/internal/telemetry/telemeter.go:58 (0x1a7ada9)
2026-01-29T23:58:58.3786786Z         	RunCommandWithOutput: return tlm.Trace(ctx, name, attrs, func(ctx context.Context) error ***
2026-01-29T23:58:58.3787671Z         /home/runner/work/terragrunt/terragrunt/internal/shell/run_cmd.go:63 (0x1a7a9f4)
2026-01-29T23:58:58.3788519Z         	RunCommandWithOutput: err := telemetry.TelemeterFromContext(ctx).Collect(ctx, "run_"+command, map[string]any***
2026-01-29T23:58:58.3789355Z         /home/runner/work/terragrunt/terragrunt/internal/tf/run_cmd.go:68 (0x1c04c51)
2026-01-29T23:58:58.3790197Z         	RunCommandWithOutput: output, err := shell.RunCommandWithOutput(ctx, l, opts, "", false, needsPTY, opts.TFPath, args...)
2026-01-29T23:58:58.3791070Z         /home/runner/work/terragrunt/terragrunt/internal/runner/run/run.go:282 (0x1fbb754)
2026-01-29T23:58:58.3792030Z         	run***WithConfig.func2: out, runTerraformError := tf.RunCommandWithOutput(childCtx, l, opts, opts.TerraformCliArgs.Slice()...)
2026-01-29T23:58:58.3793056Z         /home/runner/work/terragrunt/terragrunt/internal/runner/run/run.go:359 (0x1fbbcf1)
2026-01-29T23:58:58.3793669Z         	RunActionWithHooks: actionErrors = action(ctx)
2026-01-29T23:58:58.3794272Z         /home/runner/work/terragrunt/terragrunt/internal/runner/run/run.go:280 (0x1fbb012)
2026-01-29T23:58:58.3795183Z         	run***WithConfig: return RunActionWithHooks(ctx, l, "terraform", opts, cfg, r, func(childCtx context.Context) error ***
2026-01-29T23:58:58.3796027Z         /home/runner/work/terragrunt/terragrunt/internal/runner/run/run.go:182 (0x1fba4f6)
2026-01-29T23:58:58.3796795Z         	Run.func4: return run***WithConfig(ctx, l, opts, updated***Options, cfg, r)
2026-01-29T23:58:58.3797590Z         /home/runner/work/terragrunt/terragrunt/pkg/options/options.go:678 (0x12d455c)
2026-01-29T23:58:58.3798251Z         	(****Options).RunWithErrorHandling: err := operation()
2026-01-29T23:58:58.3798884Z         /home/runner/work/terragrunt/terragrunt/internal/runner/run/run.go:181 (0x1fba3b2)
2026-01-29T23:58:58.3799554Z         	Run: if err := opts.RunWithErrorHandling(ctx, l, r, func() error ***
2026-01-29T23:58:58.3800257Z         /home/runner/work/terragrunt/terragrunt/internal/cli/commands/run/run.go:81 (0x20e646e)
2026-01-29T23:58:58.3800924Z         	Run: return run.Run(ctx, l, tgOpts, r, runCfg, credsGetter)
2026-01-29T23:58:58.3801573Z         /home/runner/work/terragrunt/terragrunt/internal/cli/commands/run/cli.go:80 (0x20dd3e9)
2026-01-29T23:58:58.3802209Z         	NewSubcommands.func1: return Run(ctx, l, opts)
2026-01-29T23:58:58.3802824Z         /home/runner/work/terragrunt/terragrunt/internal/cli/commands/run/cli.go:69 (0x20dd3ea)
2026-01-29T23:58:58.3803485Z         	NewSubcommands.func1: return Action(l, opts)(ctx, cliCtx)
2026-01-29T23:58:58.3804158Z         /home/runner/work/terragrunt/terragrunt/internal/cli/commands/commands.go:184 (0x24e31ef)
2026-01-29T23:58:58.3804812Z         	runAction.func2: return action(actionCtx, cliCtx)
2026-01-29T23:58:58.3805422Z         /home/runner/go/pkg/mod/golang.org/x/sync@v0.19.0/errgroup/errgroup.go:93 (0x19941d0)
2026-01-29T23:58:58.3806037Z         	(*Group).Go.func1: if err := f(); err != nil ***
2026-01-29T23:58:58.3806678Z         /home/runner/.local/share/mise/installs/go/1.25.5/src/runtime/asm_amd64.s:1693 (0x48b481)
2026-01-29T23:58:58.3807627Z         	goexit: BYTE	$0x90	// NOP
2026-01-29T23:58:58.3808069Z         
2026-01-29T23:58:58.3808354Z         
2026-01-29T23:58:58.3808663Z         Stdout: (see log output above)
2026-01-29T23:58:58.3809046Z         
2026-01-29T23:58:58.3809487Z         Stderr: (see log output above)
2026-01-29T23:58:58.3810033Z     panic.go:615: Deleting test DynamoDB table terragrunt-test-locks-xpcmyo
2026-01-29T23:58:58.3810675Z     panic.go:615: Deleting test s3 bucket terragrunt-test-bucket-ykpvka
2026-01-29T23:58:58.3811202Z --- FAIL: TestAwsS3SSECustomKey (56.03s)
```

## Steps To Reproduce

Trigger CICD job

```hcl
// paste code snippets here
```

## Expected behavior

A clear and concise description of what you expected to happen.

## Must haves

- [ ] Steps for reproduction provided.

## Nice to haves

- [ ] Terminal output
- [ ] Screenshots

## Versions

- Terragrunt version:
- OpenTofu/Terraform version:
- Environment details (Ubuntu 20.04, Windows 10, etc.):

## Additional context

Add any other context about the problem here.

--- a/test/fixtures/s3-encryption/custom-key/main.tf
+++ b/test/fixtures/s3-encryption/custom-key/main.tf
@@ -1,3 +1,12 @@
+terraform {
+  required_providers {
+    null = {
+      source  = "registry.opentofu.org/hashicorp/null"
+      version = "~> 3.0"
+    }
+  }
+}
+
 resource "null_resource" "main" {}
 
 output "id" {

--- a/test/fixtures/s3-encryption/sse-aes/main.tf
+++ b/test/fixtures/s3-encryption/sse-aes/main.tf
@@ -1,3 +1,12 @@
+terraform {
+  required_providers {
+    null = {
+      source  = "registry.opentofu.org/hashicorp/null"
+      version = "~> 3.0"
+    }
+  }
+}
+
 resource "null_resource" "main" {}
 
 output "id" {

--- a/test/fixtures/s3-encryption/sse-kms/main.tf
+++ b/test/fixtures/s3-encryption/sse-kms/main.tf
@@ -1,3 +1,12 @@
+terraform {
+  required_providers {
+    null = {
+      source  = "registry.opentofu.org/hashicorp/null"
+      version = "~> 3.0"
+    }
+  }
+}
+
 resource "null_resource" "main" {}
 
 output "id" {

